cpp/src/replace/replace.cu

```diff
@@ -225,29 +333,110 @@ struct replace_kernel_forwarder {
                                            cudf::column_view const& values_to_replace,
                                            cudf::column_view const& replacement_values,
                                            rmm::mr::device_memory_resource* mr,
                                            cudaStream_t stream = 0) {
-    CUDF_FAIL("Non fixed-width types are not supported.");
+    rmm::device_scalar<cudf::size_type> valid_counter(0, stream);
+    cudf::size_type *valid_count = valid_counter.data();
+
+    auto replace_first = replace_strings_first_pass<true, false>;
+    auto replace_second = replace_strings_second_pass<true, false>;
+    if (input_col.has_nulls()) {
+      if (replacement_values.has_nulls()) {
+        replace_first = replace_strings_first_pass<true, true>;
+        replace_second = replace_strings_second_pass<true,true>;
+      }
+    } else {
+      if (replacement_values.has_nulls()) {
+        replace_first = replace_strings_first_pass<false, true>;
+        replace_second = replace_strings_second_pass<false, true>;
+      } else {
+        replace_first = replace_strings_first_pass<false, false>;
+        replace_second = replace_strings_second_pass<false, false>;
+      }
+    }
+
+    // Create new offsets column to use in kernel
+    std::unique_ptr<cudf::column> offsets = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32),
+                                                                      input_col.size() + 1,
+                                                                      cudf::mask_state::UNALLOCATED,
+                                                                      stream,
+                                                                      mr);
+    std::unique_ptr<cudf::column> indices = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32),
+                                                                      input_col.size(),
+                                                                      cudf::mask_state::UNALLOCATED,
+                                                                      stream);
+
+    auto offsets_view = offsets->mutable_view();
+    auto indices_view = indices->mutable_view();
+
+    auto device_in = cudf::column_device_view::create(input_col);
+    auto device_values_to_replace = cudf::column_device_view::create(values_to_replace);
+    auto device_replacement = cudf::column_device_view::create(replacement_values);
+    auto device_offsets = cudf::mutable_column_device_view::create(offsets_view);
+    auto device_indices = cudf::mutable_column_device_view::create(indices_view);
+
+    int valid_chars = cudf::bitmask_allocation_size_bytes(input_col.size());
+    rmm::device_buffer valid_bits(valid_chars, stream, mr);
+
+    // Call first pass kernel to get sizes in offsets
+    cudf::experimental::detail::grid_1d grid { input_col.size(), BLOCK_SIZE, 1 };
+    replace_first<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(*device_in,
```

Comment:
@jrhemstad It would be great if it would but looks like the code computes the bitmask and the offsets in the same kernel.
Honestly, I was hoping this could follow the pattern of the other strings specializations like in gather, fill, merge, etc where the bitmask is computed separately and the strings specialization code is in `strings::detail` 
But that looks like a significant rewrite here.

Reply:
Changed to use make_offsets_child_column.

Reply:
> Changed to use make_offsets_child_column.

Not pushed yet?